common/beerocks/bwl/include/bwl/nl80211_client.h

```diff
@@ -343,6 +361,22 @@ class nl80211_client {
     virtual bool get_sta_info(const std::string &interface_name, const sMacAddr &sta_mac_address,
                               sta_info &sta_info) = 0;
 
+    /**
+     * @brief Gets survey information.
+     *
+     * Survey information includes channel occupation and noise level.
+     * It is intended to periodically compute channel utilization and report AP metrics when the
+     * difference between computed value and previous value crosses a configured threshold.
+     *
+     * @param[in] interface_name Interface name, either radio or Virtual AP (VAP).
+     * @param[out] survey_info_list List of survey information structures, one for each channel,
+     * as returned by the NL80211_CMD_GET_SURVEY command.
+     *
+     * @return True on success and false otherwise.
+     */
+    virtual bool get_survey_info(const std::string &interface_name,
+                                 std::vector<survey_info> &survey_info_list) = 0;
```

Comment:
All functions in nl80211_client API return a bool to signal error and use output parameters for actual results. Newly added function just sticks to this policy to keep coherence.

I'd rather use exceptions to signal errors, but since they are not allowed in this project, I think it's better to use a bool than an empty result so the user of the API can decide whether to log an error and exit or retry after a while, for example. We cannot anticipate all the possible usages of the API, so it's better not to add limitations by design.